dll/win32/shell32/CDefView.cpp

```diff
@@ -1157,11 +1157,36 @@ LRESULT CDefView::OnCreate(UINT uMsg, WPARAM wParam, LPARAM lParam, BOOL &bHandl
         SetShellWindowEx(hwndSB, m_ListView);
     }
 
-    SHChangeNotifyEntry ntreg;
-    ntreg.fRecursive = TRUE;
-    ntreg.pidl = m_pidlParent;
+    INT nRegCount;
+    SHChangeNotifyEntry ntreg[3];
+    PIDLIST_ABSOLUTE pidls[3];
+    if ((m_FolderSettings.fFlags & FWF_DESKTOP) || _ILIsDesktop(m_pidlParent))
+    {
+        nRegCount = 3;
+        SHGetSpecialFolderLocation(m_hWnd, CSIDL_DESKTOPDIRECTORY, &pidls[0]);
+        SHGetSpecialFolderLocation(m_hWnd, CSIDL_COMMON_DESKTOPDIRECTORY, &pidls[1]);
+        SHGetSpecialFolderLocation(m_hWnd, CSIDL_BITBUCKET, &pidls[2]);
+        ntreg[0].fRecursive = TRUE;
```

Comment:
Do we need recursive watch on the desktop?
(I don't know the answer btw, just wondering)

Reply:
Non-recursive is better. OK.

Reply:
![no-error](https://user-images.githubusercontent.com/2107452/78870569-4e39c480-7a81-11ea-8028-6e3159731438.png)
Still OK.